code/src/hooks/use-filters-with-apply/deserialize-from-query-string/deserializeFromQueryString.ts

```diff
@@ -2,7 +2,7 @@ import parseSerializedRange from "@/hooks/use-filters-with-apply/parse-serialize
 import parseSerializedSet from "@/hooks/use-filters-with-apply/parse-serialized-set/parseSerializedSet";
 
 const checkNumber = (value: unknown) => {
-  return Number(value) === value;
+  return !isNaN(value as number);
 };
```

Comment:
suggestion to avoid `unknown`
```suggestion
const checkNumber = (value: string) => {
  return !isNaN(Number(value));
};
```

Reply:
I'm incredibly grateful for really useful suggestion. I applied it, looks much better now!